pkg/subctl/cmd/image_overrides/image_overrides.go

```diff
@@ -0,0 +1,77 @@
+package image_overrides
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/submariner-io/submariner-operator/pkg/versions"
+)
+
+type VersionRegistryOverrides struct {
+	version  string
+	registry string
+}
+type ImageOverrides struct {
+	operator   VersionRegistryOverrides
+	submariner VersionRegistryOverrides
+	lighthouse VersionRegistryOverrides
+}
+
+func (img *ImageOverrides) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&img.operator.registry, "operator-registry", "",
+		"Operator container registry")
+	cmd.Flags().StringVar(&img.operator.version, "operator-version", "",
+		"Operator container version")
+
+	cmd.Flags().StringVar(&img.submariner.registry, "submariner-registry", "",
+		"Submariner container registry")
+	cmd.Flags().StringVar(&img.submariner.version, "submariner-version", "",
+		"Submariner container version")
+
+	cmd.Flags().StringVar(&img.lighthouse.registry, "lighthouse-registry", "",
+		"Lighthouse container registry")
+	cmd.Flags().StringVar(&img.lighthouse.version, "lighthouse-version", "",
+		"Lighthouse container version")
+
+	_ = cmd.Flags().MarkHidden("operator-registry")
+	_ = cmd.Flags().MarkHidden("operator-version")
+	_ = cmd.Flags().MarkHidden("submariner-registry")
+	_ = cmd.Flags().MarkHidden("submariner-version")
+	_ = cmd.Flags().MarkHidden("lighthouse-registry")
+	_ = cmd.Flags().MarkHidden("lighthouse-version")
+}
+func defaultStr(defaultVal, override string) string {
+	if override != "" {
+		return override
+	} else {
+		return defaultVal
+	}
+}
+
+func (img *ImageOverrides) GetSubmarinerRegistryAndVersion(defaultRegistry, defaultVersion string) (registry, version string) {
+	defaultVersion = defaultStr(defaultVersion, versions.DefaultSubmarinerVersion)
+	registry = defaultStr(defaultRegistry, img.submariner.registry)
+	version = defaultStr(defaultVersion, img.submariner.version)
```

Comment:
Wondering why you chose this approach instead of just setting the defaults in the flag declarations?

Reply:
Hmm, that could be an option, we still need to check how it works. We have two sources of "versions", 

The --version parameter, and the specific overrides, I guess it would work... I'll try that.

Reply:
Hmm I see, though I wonder if a `--version` parameter make much sense. E.G can you deploy v0.1.0 with a v0.5.0 subctl? :)

Reply:
Shouldn’t we leave the defaults to the operator, *i.e.* use blank defaults here? (Apart from the operator...)